src/sandbox/mod.rs

```diff
@@ -0,0 +1,76 @@
+#[cfg_attr(target_os = "linux", path = "linux.rs")]
+#[cfg_attr(not(target_os = "linux"), path = "unsupported.rs")]
+mod sandbox_impl;
+
+use pyo3::{create_exception, exceptions::PyException, prelude::*, types::PyTuple};
+use std::path::PathBuf;
+
+#[derive(Clone, Debug)]
+pub enum AccessFS {
+    Read(PathBuf),
+    ReadWrite(PathBuf),
+    MakeReg(PathBuf),
+    MakeDir(PathBuf),
+}
+
+/// Enforces access restrictions
+#[pyfunction(name = "restrict_access", signature=(*rules))]
+fn py_restrict_access(rules: &PyTuple) -> PyResult<()> {
```

Comment:
We should improve on the signature of `restrict_access`. Ideally it should return a boolean value with 0 meaning it successfully restricted access, and 1 meaning the system is unsupported to be in line with the C API:

```
if (landlock_restrict_self(ruleset_fd, 0)) {
    perror("Failed to enforce ruleset");
    close(ruleset_fd);
    return 1;
}
close(ruleset_fd);
```

Reply:
or raise an exception to be more pythonic.

Reply:
It's not:

```
============================= test session starts ==============================
platform darwin -- Python 3.8.18, pytest-7.0.1, pluggy-1.0.0
rootdir: /Users/runner/work/unblob-native/unblob-native
collected 7 items

tests/test_math.py .....                                                 [ 71%]
tests/test_sandbox.py Fs                                                 [100%]

=================================== FAILURES ===================================
__________________________ test_unsupported_platform ___________________________

    @pytest.mark.skipif(platform.system() == "Linux", reason="Linux is supported.")
    def test_unsupported_platform():
        with pytest.raises(SandboxError):
>           restrict_access(AccessFS.read("/"))
E           Failed: DID NOT RAISE <class 'unblob_native.sandbox.SandboxError'>

tests/test_sandbox.py:14: Failed
------------------------------ Captured log call -------------------------------
WARNING  unblob_native.sandbox.sandbox_impl:unsupported.rs:6 Sandboxing FS access is unavailable on this system
=========================== short test summary info ============================
FAILED tests/test_sandbox.py::test_unsupported_platform - Failed: DID NOT RAI...
==================== 1 failed, 5 passed, 1 skipped in 0.08s ====================
```

Reply:
For unsupported platform I think a `SandboxError' should be raised with message `Sandboxing FS access is unavailable on this system` rather than logging the error.

Reply:
yep, only the linux codepath can raise anything currently.

Reply:
changed the code so that the non-linux codepath would also raise an exception